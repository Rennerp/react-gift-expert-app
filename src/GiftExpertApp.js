import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GiftExperApp = () => {

  const [categories, setCategories] = useState(['Samurai x']);

  return (
    <>
      <h2>GiftExperApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {
          categories.map(category =>
            <GifGrid
              category={category}
              key={category}
            />
          )
        }
      </ol>
    </>
  )
}
