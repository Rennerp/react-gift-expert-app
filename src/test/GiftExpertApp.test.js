import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import { GiftExperApp } from '../GiftExpertApp';

describe('GifExpertApp Test', () => {

  let wrapper = shallow(<GiftExperApp />);

  test('Reenderizando el componente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe mostrar la lista de categorias', () => {
    const categories = ['One punch', 'Dragon ball']
    const wrapper = shallow(<GiftExperApp defaultCategories = {categories}/>);

    expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe(categories.length);
  })

})

