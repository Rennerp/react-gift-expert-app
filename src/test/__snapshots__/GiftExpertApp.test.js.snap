// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Test Debe mostrar la lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp Test Reenderizando el componente 1`] = `
<Fragment>
  <h2>
    GiftExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
